include/deal.II/differentiation/ad.h

```diff
@@ -0,0 +1,51 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_differentiation_ad_h
+#define dealii_differentiation_ad_h
+
+#include <deal.II/base/config.h>
+
+#if defined(DEAL_II_WITH_ADOLC) || defined(DEAL_II_WITH_TRILINOS)
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
```

Comment:
@davydden As requested, here's the initial basic documentation for these namespaces. I might shuffle the documentation for `namespace Differentiation` into the module file once I expand beyond a single unifying header.